### ISSUE TITLE
no-conditional-statement and using resolve in Promise.all callback

### ISSUE THREAD

**Issue by MrZyr0** [jonaskello/eslint-plugin-functional#293]
# Bug Report

This rule is triggered when I resolve my Promise in the callback of `Promise.all()` to force me to add `return` before the `resovle()` which is not necessary.

## Expected behavior

Consider `resolve()` and `reject()` as a `return`.

## Actual behavior

EsLint tell me “Incomplete branch, every branch in a conditional statement must contain a return statement.” when I resolve my Promise in a callback of `Promise.all()`

## Steps to reproduce

```javascript
function promiseTest(returnedValue) {
	return new Promise((resolve) => {
		resolve(returnedValue);
	});
}

// eslint-disable-next-line functional/functional-parameters
function exec() {
	return new Promise((resolve) => {
		const promises = [promiseTest(), promiseTest(), promiseTest()];

		const test = new Set();

		Promise.all(promises).then((values) => {
			if (!values) {
				resolve(values);
			}

			test.add(...values);

			resolve(values);
		});
	});
}

// eslint-disable-next-line functional/no-expression-statement
exec();
```

If you add a `return` before the `resolve(values)` in the `if (!values)` the rule test pass.

## Proposed changes

Consider `resolve()` and `reject()` as a `return`.

**Comment by RebeccaStevens** [comment]
`resolve` and `reject` do not act as termination points (things can still happen after they are called).

For example:

```ts
new Promise((resolve, reject) => {
  resolve("foo");         // promise resolves with "foo".
  console.log("bar");     // "bar" will be printed despite promise already being settled.
  reject("baz");          // promise is already settled thus this call has no effect.
})
```

**Comment by MrZyr0** [comment]
Hmmm ok so should you use `return` with or after `resolve` or `reject` ?

**Comment by RebeccaStevens** [comment]
Yes

**Comment by MrZyr0** [comment]
Okay nice, it's good to know ^^

thanks
